common/plugins/org.polarsys.capella.common.helpers/src/org/polarsys/capella/common/helpers/cache/Cache.java

```diff
@@ -0,0 +1,162 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2020 THALES GLOBAL SERVICES.
+ * 
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ * 
+ * Contributors:
+ *    Thales - initial API and implementation
+ *******************************************************************************/
+package org.polarsys.capella.common.helpers.cache;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.polarsys.capella.common.helpers.Triplet;
+
+public class Cache {
+
+  private Map<Object, Object> cachedResult = new ConcurrentHashMap<>();
+
+  @SuppressWarnings("unchecked")
+  @Deprecated

```

Comment:
As a CachedFunction is a Function, we can keep this one and delete the next get(CachedFunction) one

Reply:
Done